internal/api/server.go

```diff
@@ -443,37 +444,76 @@ func (s *Server) Auth(c echo.Context) error {
 	}
 
 	if err := c.Bind(&req); err != nil {
-		return c.NoContent(http.StatusBadRequest)
+		return c.JSON(http.StatusBadRequest, map[string]string{
+			"error": "Invalid request format",
+		})
+	}
+
+	// Validate required fields
+	if err := clientutil.ValidateAuthRequest(
+		req.Message, req.Signature, req.PublicKey, req.DerivePath, req.ChainCodeHex,
+	); err != nil {
+		return c.JSON(http.StatusBadRequest, map[string]string{
+			"error": err.Error(),
```

Comment:
Please don't bleed the error detail to client side

Reply:
pushed the fix